AdGoBye/Indexer.cs

```diff
@@ -18,8 +18,39 @@ public class Indexer
     public static void ManageIndex()
     {
         using var db = new State.IndexContext();
-        if (db.Content.Any()) VerifyDbTruth();
-        var contentFolders = new DirectoryInfo(GetCacheDir()).GetDirectories();
+        const int maxRetries = 3;
+        const int delayMilliseconds = 5000;
+
+        DirectoryInfo[]? contentFolders = null;
+        for (var retry = 0; retry < maxRetries; retry++)
+        {
+            try
+            {
+                contentFolders = new DirectoryInfo(GetCacheDir()).GetDirectories();
+                break;
+            }
+            catch (DirectoryNotFoundException ex)
+            {
+                // print exception with exception message if it's the last retry
+                if (retry == maxRetries - 1)
+                {
+                    Logger.Error(ex, $"Directory not found attempting retry: {retry + 1} of {maxRetries}: Directory not found. No more retries.");
+                }
+                else
+                {
+                    Logger.Error($"Directory not found attempting retry: {retry + 1} of {maxRetries}");
+                }
+                Thread.Sleep(delayMilliseconds); 
+            }
+        }
+        
+        if (contentFolders == null)
+        {
+            Logger.Error("Max retries reached. Unable to find directory please define it manually in appsettings.json, closing in 5 seconds.");
+            Thread.Sleep(delayMilliseconds);
+            Environment.Exit(0);
+        }
+        
```

Comment:
This code seems weird to me, `ManageIndex` is fairly early in execution and us not managing to find the Cache directory is a fatal failure state. You're also exiting with `0` which is a success exit code.

I'd say move the logic for exiting into the no more retries block and let Serilog deconstruct and rethrow the exception (by using `Logger.Fatal`, essentially what [SteamParser](https://github.com/AdGoBye/AdGoBye/blob/3814f9e3f2dcf3fd65dfc791c3f553c62837681a/AdGoBye/SteamParser.cs#L32-L36) does) rather than essentially logging the same failure twice. 

I'm iffy about the Thread.Sleep before the exit because I think the responsibility of leaving the output open is for the shell rather than us, I'm fine with leaving in but the pattern I described probably would look very weird this way.

Reply:
opted to remove the sleep as the retry attempts should be more than enough to get a users attention.